scripts/documentation/tsdoc.ts

```diff
@@ -0,0 +1,361 @@
+   * @returns Promise<{}>
+   */
+  async generateJSONFromFiles() {
+    if (!this.templateFunction) console.log('AHHHHHH MEOW')
+    return await new Documentalist()
+      .use(/\.ts$/, new TypescriptPlugin({
+        excludePaths: ['node_modules'],
+        includeDeclarations: true,
+      }))
+      .documentGlobs('packages/**/*')
+      .catch((error) => console.error(error)); // tslint:disable-line
+  }
+
+  /**
+   * The main function of this class. Iterates through all modules found in docData (json).
+   * This should already be generated from this.generateJSONFromFiles().
+   * @param docData json containing documentation from documentalist
+   */
+  generateDocs(docData) {
+    if (!docData) {
+      console.error('FAILURE: Documentation generation did not compile correctly.'); // tslint:disable-line
+    }
+    const markdownBuffer: MarkdownBuffer = {};
+
+    this.docData = docData.typescript;
+    Object.keys(this.docData).forEach((module) => {
+      console.log(`-- generating docs for ${module}`); // tslint:disable-line
+      const docs = this.generateDocsForModule(module);
+      if (docs) {
+        const {readmeDirectoryPath, moduleName} = docs;
+        const markdownComponentBuffer = markdownBuffer[readmeDirectoryPath];
+        const isComponent =
+          !moduleName.toLowerCase().includes(FOUNDATION) && !moduleName.toLowerCase().includes(ADAPTER);
+
+        if (markdownComponentBuffer) {
+          if (isComponent) {
+            markdownComponentBuffer.component.push(docs);
+          } else {
+            markdownComponentBuffer.nonComponent.push(docs);
+          }
+        } else {
+          if (isComponent) {
+            markdownBuffer[readmeDirectoryPath] = {component: [docs], nonComponent: []};
+          } else {
+            markdownBuffer[readmeDirectoryPath] = {component: [], nonComponent: [docs]};
+          }
+        }
+      }
+    });
+
+    this.generateMarkdownFileFromBuffer(markdownBuffer);
+  }
+
+  /**
+   * Creates documentation for a specified `esmodule`.
+   * @param esmodule module name (ie. MDCSelectIconFoundation)
+   */
+  generateDocsForModule(esmodule: string): ModuleMarkdown | undefined {
+    const {kind, fileName} = this.docData[esmodule];
+    const readmeDirectoryPath = this.getFilePathName(fileName);
+
+    if (kind === 'type alias') {
+      // ignore types and interfaces
+      return;
+    }
+    return {
+      events: this.getDocumentationForModule(esmodule).events,
+      methods: this.getDocumentationForMethods(esmodule),
+      moduleName: esmodule,
+      properties: this.getDocumentationProperties(esmodule),
+      readmeDirectoryPath,
+    };
+  }
+
+  /**
+   * This is higher level documentation from the class.
+   * Currently this should only include events documentation.
+   * @returns documentation of the esmodule.
+   * @param esmodule module name (ie. MDCSelectIconFoundation)
+   */
+  getDocumentationForModule(esmodule: string): ModuleDocumentation {
```

Comment:
Can we merge this into `getDocumentationForEvents`?

Reply:
I was thinking if we ever wanted another tag off of the module, that it would be easier to split like this. But I can combine this into one for now, since there isn't anything we need.